### PR TITLE
[scheduler] Prevent frequent updates for unassigned shoots

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area robustness
/kind enhancement

**What this PR does / why we need it**:
This PR contributes the following improvements to the scheduler:
- Sort error message of non-eligible seed candidates to prevent status flapping.
- The scheduler not to be triggered on shoot status updates. This circumvented the reconciliation backoff in the past.

**Special notes for your reviewer**:
/cc @ScheererJ thanks for reporting

**Release note**:

```noteworthy operator
The `gardener-scheduler` was improved to consider reconciliation backoffs. In the past, unassigned shoots were affected by frequent scheduler reconciliations and status updates which potentially strained the scheduler and etcd.
```

### CODE REVIEW

**Review by LucaBernstein — COMMENTED** [review]
/lgtm
Nice one, thank you!

File: pkg/scheduler/controller/shoot/reconciler.go [hunk 1]
```diff
@@ -329,34 +331,34 @@ func applyStrategy(log logr.Logger, shoot *gardencorev1beta1.Shoot, seedList []g
 
 func filterCandidates(shoot *gardencorev1beta1.Shoot, shootList []*gardencorev1beta1.Shoot, seedList []gardencorev1beta1.Seed) ([]gardencorev1beta1.Seed, error) {
 	var (
-		candidates      []gardencorev1beta1.Seed
-		candidateErrors = make(map[string]error)
-		seedUsage       = v1beta1helper.CalculateSeedUsage(shootList)
+		candidates []gardencorev1beta1.Seed
+		seedToErr  = make(map[string]error)
```
Comment (rfranzke) [inline pkg/scheduler/controller/shoot/reconciler.go]:
```suggestion
		seedNameToErr  = make(map[string]error)
```

**Review by rfranzke — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**